src/wazuh_modules/task_manager/wm_task_manager_db.c

```diff
@@ -280,4 +284,85 @@ int wm_task_manager_update_task_status(int agent_id, const char *module, const c
     return WM_TASK_SUCCESS;
 }
 
+int wm_task_manager_get_task_by_agent_id_and_module(int agent_id, const char *module, char **command, char **status, int *create_time, int *last_update_time) {
+    sqlite3 *db = NULL;
+    sqlite3_stmt *stmt = NULL;
+    int result = OS_INVALID;
+
+    if (sqlite3_open_v2(TASKS_DB, &db, SQLITE_OPEN_READWRITE, NULL)) {
+        mterror(WM_TASK_MANAGER_LOGTAG, MOD_TASK_OPEN_DB_ERROR);
+        return wm_task_manager_sql_error(db, stmt);
+    }
+    if (wdb_prepare(db, task_queries[WM_TASK_GET_TASK_BY_AGENT_ID_AND_MODULE], -1, &stmt, NULL) != SQLITE_OK) {
+        mterror(WM_TASK_MANAGER_LOGTAG, MOD_TASK_SQL_PREPARE_ERROR);
+        return wm_task_manager_sql_error(db, stmt);
+    }
+    int task_id;
+    sqlite3_bind_text(stmt, 1, module, -1, NULL);
+    sqlite3_bind_int(stmt, 2, agent_id);
+
+    if (result = wdb_step(stmt), result != SQLITE_ROW) {
+        mterror(WM_TASK_MANAGER_LOGTAG, MOD_TASK_SQL_STEP_ERROR);
+        return wm_task_manager_sql_error(db, stmt);
+    }
+
+    task_id = sqlite3_column_int(stmt, 0);
+
+    if (!task_id) {
+        wdb_finalize(stmt);
+        sqlite3_close_v2(db);
+        return OS_NOTFOUND;
+    }
+    sqlite_strdup((char*)sqlite3_column_text(stmt, 1), *command);
```

Comment:
Add this in an else so you can remove lines 312 and 313.

Reply:
done